lte/gateway/c/oai/tasks/sgw/pgw_handlers.c

```diff
@@ -536,82 +549,137 @@ uint32_t pgw_handle_nw_initiated_bearer_deactv_req(
-  }
 
-  itti_s5_deactv_ded_bearer_req =
-    &message_p->ittiMsg.s5_nw_init_deactv_bearer_request;
-  //Send ITTI message to SGW
-  memset(
-    itti_s5_deactv_ded_bearer_req,
-    0,
-    sizeof(itti_s5_nw_init_deactv_bearer_request_t));
-  itti_s5_deactv_ded_bearer_req->delete_default_bearer = false;
-  itti_s5_deactv_ded_bearer_req->no_of_bearers = bearer_req_p->no_of_bearers;
-  memcpy(
-    &itti_s5_deactv_ded_bearer_req->ebi,
-    bearer_req_p->ebi,
-    (sizeof(ebi_t)) * bearer_req_p->no_of_bearers);
+
   hashtblP = spgw_state->sgw_state.s11_bearer_context_information;
   if (hashtblP == NULL) {
     OAILOG_ERROR(
       LOG_PGW_APP, "hashtblP is NULL for nw_initiated_deactv_bearer_req\n");
     OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
   }
 
-  //Check if EBI recvd == LBI to know if default bearer has to be deactivated
+  // Check if valid LBI and EBI recvd
   while ((num_elements < hashtblP->num_elements) && (i < hashtblP->size) &&
-         (!found)) {
+         (!is_imsi_found)) {
     pthread_mutex_lock(&hashtblP->lock_nodes[i]);
     if (hashtblP->nodes[i] != NULL) {
       node = hashtblP->nodes[i];
       spgw_ctxt_p = node->data;
       num_elements++;
       if (spgw_ctxt_p != NULL) {
         if (!strcmp(
-              (const char *)
-                spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
-              (const char *) bearer_req_p->imsi)) {
-          itti_s5_deactv_ded_bearer_req->s11_mme_teid =
+          (const char *)
+          spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
+          (const char *) bearer_req_p->imsi)) {
+          is_imsi_found = true;
+          s11_mme_teid =
             spgw_ctxt_p->sgw_eps_bearer_context_information.mme_teid_S11;
-          for (j = 0; j < bearer_req_p->no_of_bearers; j++) {
-            if (
-              bearer_req_p->ebi[j] ==
-              spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
-                .default_bearer) {
-              itti_s5_deactv_ded_bearer_req->delete_default_bearer = true;
-              found = true;
-              break;
+          if ((bearer_req_p->lbi != 0) &&
+            (bearer_req_p->lbi ==
+            spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
+              .default_bearer)) {
+            is_lbi_found = true;
+            // Check if the received EBI is valid
+            for (itrn = 0; itrn < bearer_req_p->no_of_bearers; itrn ++) {
+              if(sgw_cm_get_eps_bearer_entry(
+                &spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection,
+                bearer_req_p->ebi[itrn])) {
+                is_ebi_found = true;
+                ebi_to_be_deactivated[no_of_bearers_to_be_deact] =
+                  bearer_req_p->ebi[itrn];
+                no_of_bearers_to_be_deact++;
+              } else {
+                invalid_bearer_id[no_of_bearers_rej] = bearer_req_p->ebi[itrn];
+                no_of_bearers_rej++;
```

Comment:
if we have multiple EBIs, there is a chance that some are found and some are not. What should be the behaviour in such a case?